include/drivers/flash.h

```diff
@@ -289,6 +297,69 @@ typedef bool (*flash_page_cb)(const struct flash_pages_info *info, void *data);
 void flash_page_foreach(struct device *dev, flash_page_cb cb, void *data);
 #endif /* CONFIG_FLASH_PAGE_LAYOUT */
 
+#if defined(CONFIG_FLASH_JESD216_API) || defined(DT_DOXYGEN)
```

Comment:
This is not how we do it. Please add CONFIG_FLASH_JESD216_API to doc/zephyr.doxyfile.in if you want this documented or avoid the #ifdef in headers.

What does the DT stand for exactly?

Reply:
The flag was originally added to support devicetree definitions.